cmd/cluster/transferowner.go

```diff
@@ -83,6 +89,68 @@ func generateServiceLog(clusterId string, template string) servicelog.PostCmdOpt
 	}
 }
 
+func updateManifestWork(conn *sdk.Connection, kubeCli client.Client, clientSet *kubernetes.Clientset, clusterID, mgmtClusterName string, pullsecret []byte) error {
+
+	if err := workv1.AddToScheme(kubeCli.Scheme()); err != nil {
+		return fmt.Errorf("failed to add scheme: %w", err)
+	}
+
+	manifestWorkName := clusterID
+	manifestWorkNamespace := mgmtClusterName
+	hostedCluster, err := utils.GetClusterAnyStatus(conn, clusterID)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster: %w", err)
+	}
+
+	secretName := hostedCluster.Name() + "-pull"
+
+	manifestWork := &workv1.ManifestWork{}
+	err = kubeCli.Get(context.TODO(), types.NamespacedName{Name: manifestWorkName, Namespace: manifestWorkNamespace}, manifestWork)
+	if err != nil {
+		return fmt.Errorf("failed to get the target manifestwork for given cluster %v: %w", clusterID, err)
+	}
+
+	for i, manifest := range manifestWork.Spec.Workload.Manifests {
+		if manifest.Raw != nil {
+			var manifestData map[string]interface{}
+			err := json.Unmarshal(manifest.Raw, &manifestData)
+			if err != nil {
+				return err
+			}
+
+			if manifestData["kind"] == "Secret" {
+				jsonData, err := json.Marshal(manifestData)
+				if err != nil {
+					return err
+				}
+				secret := &corev1.Secret{}
+				err = json.Unmarshal(jsonData, secret)
+				if err != nil {
+					return err
+				}
+				if secret.Name == secretName {
+					secret.Data[".dockerconfigjson"] = pullsecret
```

Comment:
The pull secret name also needs an update to ensure that the new secret is populated in nodes. We can maybe generate a random suffix at the end
We also need to be aware of clusters that have different domain name

Reply:
I am not quite understand about this one, I think the secret name is fixed as the `${clustername}-pull` and the secret data key name is fixed as `.dockerconfigjson`
So we just do updating the existing secret key with the new value.

Reply:
Please refer to Note in step 5 of https://github.com/openshift/ops-sop/blob/master/hypershift/knowledge_base/howto/replace-pull-secret.md#replaceupdate-current-pull-secret

Reply:
ok, thanks, seems there are several more steps to replace the pull secret, I will update the core logic to follow the same